typings/index.d.ts

```diff
@@ -14,9 +14,8 @@
 /// <reference path="javascript.d.ts" />
 /// <reference path="msg/msg.d.ts" />
 
-import * as Blockly from './core';
-import './blocks';
-import './javascript';
+import { Generator } from 'core/blockly';
+export * from './core';
+export * from './blocks';
+export const JavaScript: Generator;
```

Comment:
1. Should we provide typings for the other language generators too?  They do all get loaded if you import `'blockly/blocks'` in node.js, even if they don't in the browser (ugh).
0. I believe the actual typings for generator module default exports are considerably more elaborate than just a simple `Generator` instance—for example, `generators/javascript.js` adds a bunch of `.ORDER_*` properties that are not on base `Generator` instances.  Do we care about such things?  ("It would be nice but is not important at the moment" would be a perfectly reasonable reply.)

Reply:
1. The generators are loaded if you import `blockly/blocks`? why / where is this done? I don't see anything about it in the various `blocks`-related files... why would we do this and do we intend to keep doing it? in any case then I guess we should, but they'd have to go into the blocks definition file, which gets complicated as then you would have to reference the generators off of the import object from the blocks which makes very little sense imo.
2. Yeah, I know there's extra stuff for them but we don't have existing type definitions for the extra stuff and I don't have time to add them now. I think this will be resolved when we just move everything to TS.

Reply:
Okay I see in `node.js` where we import all of the generators, so I'm going to assume you had a typo and meant "if you import `blockly' in node.js". In that case then yes I can add type definitions for the other generators to the index.d.ts, but also the fact that we do that is contrary to what our documentation says... do you think we plan to keep doing that? if we don't then i'd rather not add the definitions here because i don't want to encourage people to rely on these secretly imported modules we haven't previously told anyone about.